### PR TITLE
[Testing]: added tests for Accommodation layer

### PR DESCRIPTION
**Pull request by vadymhrnk**
##  Added:
- AccommodationControllerTest
  - getById_ValidAccommodation_Successful()
- AccommodationRepositoryTest
  - findById_ValidAccommodation_ShouldReturnOptionalAccommodation()
- AccommodationServiceImplTest
  - save_ValidAccommodation_Successful()
- SQL Scripts
  - add-accommodations.sql
  - delete-accommodations.sql
  - add-accommodations-amenities.sql
  - delete-accommodations-amenities.sql

### CODE REVIEW

File: src/test/java/com/example/accommodationbookingapp/repository/accommodation/AccommodationRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        Long accommodationId = 1L;
+
+        Optional<Accommodation> result = accommodationRepository.findById(accommodationId);
+
+        Assertions.assertTrue(result.isPresent());
+        Accommodation actual = result.get();
+
+        Accommodation expected = accommodation;
+
+        Assertions.assertTrue(EqualsBuilder.reflectionEquals(expected, actual, "id"));
```
Comment (annton-shakur) [inline src/test/java/com/example/accommodationbookingapp/repository/accommodation/AccommodationRepositoryTest.java:79]: just using EqualsBuilder.reflectionEquals(expected, actual, "id") should work here as well, I believe

File: src/test/java/com/example/accommodationbookingapp/service/accommodation/impl/AccommodationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        when(accommodationMapper.toResponseDto(accommodationModel))
+                .thenReturn(accommodationResponseDto);
+
+        AccommodationResponseDto expectedDto = accommodationResponseDto;
+
+        AccommodationResponseDto actualDto = accommodationServiceImpl
+                .save(createAccommodationRequestDto);
+
+        Assertions.assertNotNull(actualDto);
+        Assertions.assertTrue(EqualsBuilder.reflectionEquals(expectedDto, actualDto, "id"));
```
Comment (annton-shakur) [inline src/test/java/com/example/accommodationbookingapp/service/accommodation/impl/AccommodationServiceImplTest.java:84]: here as well

**Review by annton-shakur — APPROVED** [review]
Good job!

**Review by TarasYurkovskyi — APPROVED** [review]
GJ!

File: src/test/java/com/example/accommodationbookingapp/repository/accommodation/AccommodationRepositoryTest.java (hunk 1 above)
Comment (vadymhrnk) [reply src/test/java/com/example/accommodationbookingapp/repository/accommodation/AccommodationRepositoryTest.java:79]: Yeah, I also had the same thought. But when I actually used a debug on one of my methods, it was two completely different objects.
assertTrue is needed here

**Pull request merged**